### PR TITLE
switch to mainline immutable rc-10

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] type image stubs so we can require images as strings correctly
- [x] fix some wallet typing
- [x] fix some fs typing
- [x] change to OrderedSet and just sort.

We do lose partialSort and SortedSet but i think we should likely handle this differently. Instead we should just keep the newest 10 messages and keep that in the store. then the remote and inbox can just use that and not have to sort all the time

### CODE REVIEW

File: shared/constants/fs.js [hunk 1]
```diff
@@ -602,7 +602,10 @@ export const isMedia = (pathItem: Types.PathItem): boolean =>
   pathItem.type === 'file' && ['image', 'av'].includes(viewTypeFromMimeType(pathItem.mimeType))
 
 const slashKeybaseSlashLength = '/keybase/'.length
-export const generateFileURL = (path: Types.Path, localHTTPServerInfo: ?Types._LocalHTTPServer): string => {
+export const generateFileURL = (
+  path: Types.Path,
+  localHTTPServerInfo: ?$ReadOnly<Types._LocalHTTPServer>
```
Comment (chrisnojima) [inline shared/constants/fs.js]: we should avoid exposing these _ types actually but this is the smallest change that fixes this for now

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready
cc: @jzila @songgao

**Comment by MarcoPolo** [comment]
🔥 all the `// $FlowIssue`

**Comment by chrisnojima** [comment]
yeah!

**Pull request merged**